hconf.c

```diff
@@ -149,6 +156,77 @@ hconf_input_mode_handler(SYSCTL_HANDLER_ARGS)
         return (0);
 }
 
+static int
+hconf_get_switch(struct hconf_softc *sc, int swtype, u_int *mask)
+{
+	uint8_t *fbuf;
+	int error;
+
+	if (sc->switch_rlen[swtype] <= 1)
+		return (ENXIO);
+
+	fbuf = malloc(sc->switch_rlen[swtype], M_TEMP, M_WAITOK | M_ZERO);
+	sx_xlock(&sc->lock);
```

Comment:
hid requests are serialized by transport backends now, so no need to protect them with extra locks.  hconf_set_input_mode_impl() takes sx_lock to serialize soft context updates.